src/compas_fab/artists/base.py

```diff
@@ -129,88 +130,54 @@ def scale_link(self, link, transformation):
             # Recursive call
             self.scale_link(child_joint.child_link, transformation)
 
-    def update(self, configuration, names, visual=True, collision=True):
-        """Trigger the update of the robot geometry.
+    def _apply_transformation_on_transformed_link(self, item, transformation):
+        """Applies a transformation on a link that is already transformed.
+
+        Calculates the relative transformation and applies it to the link
+        geometry. This is to prevent the recreation of large meshes.
 
         Parameters
         ----------
-        configuration : :class:`compas_fab.robots.Configuration`
-            Instance of the configuration (joint state) to move to.
-        collision : bool, optional
-            ``True`` if the collision geometry should be also updated, otherwise ``False``.
-            Defaults to ``True``.
+        item: :class:`compas.robots.Visual` or :class:`compas.robots.Collision`
+            The visual or collidable object of a link.
+        transformation: :class:`Transformation`
+            The (absolute) transformation to apply onto the link's geometry.
+
+        Returns
+        -------
+        None
         """
-        positions = configuration.values
-        self.update_links(names, positions, visual, collision)
+        relative_transformation = transformation * item.current_transformation.inverse()
+        self.transform(item.native_geometry, relative_transformation)
+        item.current_transformation = transformation
 
-    def update_links(self, names, positions, visual=True, collision=True):
-        """Updates the joints and link geometries.
+    def update(self, configuration, names, collision=True):
```

Comment:
One more comment: adding the `names` parameter will feels slightly odd here. I assume this is to support partial configurations, but in general, we do this by passing a `group` parameter if the values only apply to a sub-set of the joints.

Could we replace it with `group`, and then if `group is None` we set `names` as default to `robot.get_configurable_joint_names()`?

Reply:
This is the problem of having a divide between robot semantics and robot model. Only semantics knows about the joint names that belong to the planning group. So they are now passed from the `compas_fab.robots.Robot` over the artist to the model. 
One option would be to move semantics as attribute to the robot model (as discussed once....)

Reply:
After discussion, made a card:  https://github.com/compas-dev/compas_fab/projects/1#card-27015436 to be handled in another PR